pkg/pgmodel/migrate.go

```diff
@@ -58,101 +71,251 @@ func (t *mySrc) replaceSchemaNames(r io.ReadCloser) (io.ReadCloser, error) {
+		return fmt.Errorf("unable to set version in DB: %w", err)
+	}
+
+	return nil
 }
 
 var migrateMutex = &sync.Mutex{}
 
-// Migrate performs a database migration to the latest version
-func Migrate(db *sql.DB, versionInfo VersionInfo) (err error) {
-	migrateMutex.Lock()
-	defer migrateMutex.Unlock()
-	// The migration table will be put in the public schema not in any of our schema because we never want to drop it and
-	// our scripts and our last down script drops our shemas
-	driver, err := postgres.WithInstance(db, &postgres.Config{MigrationsTable: "prom_schema_migrations"})
+func execMigrationFiles(tx pgx.Tx, name string) error {
+	f, err := migrations.MigrationFiles.Open(name)
 	if err != nil {
-		return fmt.Errorf("cannot create driver due to %w", err)
+		return fmt.Errorf("unable to get migration scripts: name %s, err %w", name, err)
 	}
 
-	_, err = db.Exec(timescaleInstall)
-	if err != nil {
-		return fmt.Errorf("timescaledb failed to install due to %w", err)
+	var (
+		entries []os.FileInfo
+		entry   os.FileInfo
+		file    http.File
+	)
+
+	if myToC, ok := toc[name]; ok {
+		entries = make([]os.FileInfo, 0, len(myToC))
+		for _, fileName := range myToC {
+			fullName := filepath.Join(name, fileName)
+			file, err = migrations.MigrationFiles.Open(fullName)
+			if err != nil {
+				return fmt.Errorf("unable to get migration script from toc: name %s, err %w", fullName, err)
+			}
+
+			if entry, err = file.Stat(); err != nil {
+				return fmt.Errorf("unable to stat migration script from toc: name %s, err %w", fullName, err)
+			}
+
+			entries = append(entries, entry)
+		}
+	} else {
+		entries, err = f.Readdir(-1)
+		if err != nil {
+			return fmt.Errorf("unable to read migration scripts directory: name %s, err %w", name, err)
+		}
+
+		sort.Slice(entries, func(i, j int) bool {
+			first, err := strconv.Atoi(strings.Split(entries[i].Name(), "-")[0])
+			if err != nil {
+				return false
+			}
+			second, err := strconv.Atoi(strings.Split(entries[j].Name(), "-")[0])
+			if err != nil {
+				return false
+			}
+
+			return first < second
+		})
+	}
+
+	for _, e := range entries {
+		// Ignoring directories.
+		if e.IsDir() {
+			continue
+		}
+		fileName := filepath.Join(name, e.Name())
+		f, err := migrations.MigrationFiles.Open(fileName)
+		if err != nil {
+			return fmt.Errorf("unable to get migration script: name %s, err %w", fileName, err)
+		}
+		contents, err := replaceSchemaNames(f)
+		if err != nil {
+			return fmt.Errorf("unable to read migration script: name %s, err %w", fileName, err)
+		}
+		_, err = tx.Exec(context.Background(), string(contents))
+		if err != nil {
+			return fmt.Errorf("error executing migration script: name %s, err %w", fileName, err)
+		}
 	}
 
-	src, err := httpfs.New(migrations.SqlFiles, "/")
+	return nil
+}
+
+func installVersionScripts(tx pgx.Tx, dbVersion, appVersion semver.Version) error {
```

Comment:
is this installing scripts up to the supplied version, greater than the supplied version or between them? Either way, it should be reflected in the naming: either something like
`installScriptsOlderThan`, something like `installScriptsNewerThan` or `installScriptsBetween`